regulations/static/components/Collapsible.vue

```diff
@@ -40,54 +44,61 @@ export default {
             required: false,
             default: "1s",
         },
-        direction: { //horizontal or vertical
+        direction: {
+            //horizontal or vertical
             type: String,
             required: true,
         },
     },
 
-    data: function() {
+    data: function () {
         return {
             size: 0,
             visible: true,
             isVertical: true,
             styles: {
                 overflow: "hidden",
                 transition: this.transition,
-            }
-        }
+            },
+        };
     },
 
     computed: {
-        sizeStyle: function() {
-            return this.isVertical ? 
-                { height: this.visible ? this.size : 0 } :
-                { width: this.visible ? this.size : 0 };
-        }
+        sizeStyle: function () {
+            return this.isVertical
+                ? { height: this.visible ? this.size : 0 }
+                : { width: this.visible ? this.size : 0 };
+        },
     },
 
     methods: {
-        resize: function(e) {
+        resize: function (e) {
             this.computeSize();
         },
-        toggle: function(target) {
-            if(this.name === target) {
-                this.visible = !this.visible;
+        toggle: function (target) {
+            if (this.name === target) {
+                if (!this.visible) {
+                    this.computeSize();
+                }
+                requestAnimationFrame(() => {
+                    this.visible = !this.visible;
+                });
             }
         },
-        computeSize: function() {
+        computeSize: function () {
             let setProps = (visibility, display, position, size) => {
```

Comment:
These two functions should be able to be pulled out now to the level of methods like `computeSize`